nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -1475,10 +1476,19 @@ public Description matchVariable(VariableTree tree, VisitorState state) {
     if (initializer != null) {
       if (!symbol.type.isPrimitive() && !skipDueToFieldAnnotation(symbol)) {
         if (mayBeNullExpr(state, initializer)) {
-          ErrorMessage errorMessage =
-              new ErrorMessage(
-                  MessageTypes.ASSIGN_FIELD_NULLABLE,
-                  "assigning @Nullable expression to @NonNull field");
+          ErrorMessage errorMessage;
+          if (hasNestedClass(state.getTypes(), symbol.type)) {
+            errorMessage =
+                new ErrorMessage(
+                    MessageTypes.NULLABLE_ON_WRONG_NESTED_CLASS_LEVEL,
+                    "Type-use nullability annotations should be applied on inner class");
```

Comment:
I can't understand this check and why it occurs here?  I think the check should occur independent of whether the initializer assignment is valid.  If we need to possibly report multiple error messages from this method, we can use the `state.reportMatch` API to report the errors and then just always return `Description.NO_MATCH`.

Reply:
Changed the logic here to adhere to the [ErrorProne flow](https://github.com/google/error-prone/blob/master/core/src/main/java/com/google/errorprone/bugpatterns/NullableOnContainingClass.java).